source/material_model/rheology/composite_visco_plastic.cc

```diff
@@ -183,22 +187,31 @@ namespace aspect
         double strain_rate_residual = 2*strain_rate_residual_threshold;
         double strain_rate_deriv = 0;
         unsigned int stress_iteration = 0;
+        std::pair<double, double> creep_edot_and_deriv;
+
         while (std::abs(strain_rate_residual) > strain_rate_residual_threshold
                && stress_iteration < stress_max_iteration_number)
           {
 
-            const std::pair<double, double> creep_edot_and_deriv = compute_strain_rate_and_derivative (creep_stress,
-                                                                   pressure,
-                                                                   temperature,
-                                                                   composition,
-                                                                   diffusion_creep_parameters,
-                                                                   dislocation_creep_parameters,
-                                                                   peierls_creep_parameters,
-                                                                   drucker_prager_parameters);
-
-            const double strain_rate = creep_stress/(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*creep_edot_and_deriv.first;
+            creep_edot_and_deriv = compute_creep_strain_rate_and_derivative (creep_stress,
+                                                                             pressure,
+                                                                             temperature,
+                                                                             composition,
+                                                                             diffusion_creep_parameters,
+                                                                             dislocation_creep_parameters,
+                                                                             peierls_creep_parameters,
+                                                                             drucker_prager_parameters);
+
+            double strain_rate = creep_stress/(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*creep_edot_and_deriv.first;
             strain_rate_deriv = 1./(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*creep_edot_and_deriv.second;
 
+            if (use_elasticity)
```

Comment:
Great that this all that was needed (for now) to incorporate elasticity! Certainly easier when elasticity is out of the "creep" circuit.